### ISSUE TITLE
ComboBox focused last

### ISSUE THREAD

**Issue by nanocortex** [migueldeicaza/gui.cs#742]
Hey, maybe I do something wrong but I have a problem that ComboBox is always focused last.

![8SijcLCnp8](https://user-images.githubusercontent.com/40443396/86482710-c8ad7400-bd52-11ea-807c-93c56da4ebb9.gif)

```csharp
using System.Collections.Generic;
using NStack;
using Terminal.Gui;

namespace UICatalog.Scenarios {
	[ScenarioMetadata (Name: "ComboBox focus bug",
		Description: "Demonstrates ComboBox bug")]
	[ScenarioCategory ("Bug Repro")]
	public class ComboBoxBug : Scenario {
		public override void Setup ()
		{
			var typeRadioGroup = new RadioGroup (0, 0, new ustring [] {"Expense", "Income"}, 0);
			Win.Add (typeRadioGroup);

			var nameLabel = new Label ("Name") {
				X = Pos.Left (typeRadioGroup),
				Y = Pos.Top (typeRadioGroup) + 2
			};
			var nameTextField = new TextField ("") {
				X = Pos.Left (nameLabel),
				Y = Pos.Top (nameLabel) + 1,
				Width = Dim.Fill ()
			};
			Win.Add (nameLabel);
			Win.Add (nameTextField);

			var categoryLabel = new Label ("Category ComboBox") {
				X = Pos.Left (nameTextField),
				Y = Pos.Top (nameTextField) + 2
			};

			var categoryComboBox = new ComboBox () {
				ColorScheme = Colors.TopLevel,
				X = Pos.Left (categoryLabel),
				Y = Pos.Top (categoryLabel) + 1,
				Width = Dim.Fill (),
				Height = Dim.Fill (2),
			};

			categoryComboBox.SetSource (new List<string> () {
				"Cat 1",
				"Other category"
			});
			
			Win.Add (categoryLabel);
			Win.Add (categoryComboBox);
			
			var amountLabel = new Label ("Amount") {
				X = Pos.Left (categoryComboBox),
				Y = Pos.Top (categoryComboBox) + 2
			};
			var amountTextField = new TextField ("") {
				X = Pos.Left (amountLabel),
				Y = Pos.Top (amountLabel) + 1,
				Width = Dim.Fill ()
			};
			Win.Add (amountLabel);
			Win.Add (amountTextField);
			
			var saveButton = new Button ("Save") {
				X = Pos.Left (amountTextField),
				Y = Pos.Top (amountTextField) + 2
			};

			var cancelButton = new Button ("Cancel") {
				X = Pos.Right (saveButton),
				Y = Pos.Top (saveButton),
				Clicked = Application.RequestStop
			};

			Win.Add (saveButton);
			Win.Add (cancelButton);
		}
	}
}
```

**Comment by BDisp** [comment]
I haven't tested it yet but what I think about this behavior is due to the `ComboBox `being composed of a `TextField `and a `ListView`. As the `TextField `is created dynamically after the construction of the class and is what is visible naturally, it must be its index that is considered.

**Comment by nanocortex** [comment]
I'm new to this codebase (by the way, nice work!), but I traced down that this line is causing a problem:

https://github.com/migueldeicaza/gui.cs/blob/527c8aa36526e0e0347cecae7ecd66196ee5aae4/Terminal.Gui/Views/ComboBox.cs#L333

It's after that operation `ComboBox` is moved to the end of `subviews` list.

**Comment by BDisp** [comment]
Thank you for my part. It was what I suspected. This method removes a view and adds it on top of the other views and its index is changed. We will have to force it to maintain the same index. Thanks for the info.

**Comment by fergusonr** [comment]
I cannot reproduce the issue with the latest from PR #710

**Comment by BDisp** [comment]
> 
> 
> I cannot reproduce the issue with the latest from PR #710

It doesn't really happen anymore. Would it have been because it was changed from
```cs
listview.Redraw (new Rect (0, 0, width, height)); // for any view behind this
```
to this?
```cs
listview.Clear (); // Ensure list shrinks in Dialog as you type
```

**Comment by fergusonr** [comment]
Shift + Tab fixed in PR #710
